thunder/distributed/transforms/fsdp_v2.py

```diff
@@ -263,6 +259,36 @@ def transform_state_dict_for_submodule(
             new_state_dict[k] = v
         return new_state_dict
 
+    def reverse_transform_state_dict_for_submodule(
+        self,
+        model: ThunderModule,
+        submodule_name: str,
+        state_dict: dict[str, Any],
+    ) -> dict[str, Any]:
+        from thunder.executors.torchex import _all_gather_prim_impl
+
+        for name, tensor in state_dict.items():
```

Comment:
I wonder if we should make a new dict (never thought about the implications, though, so just a gut feeling).

Reply:
My take is that reverse transform is closed in `original_state_dict` so it wouldn't matter a lot whether creating a new dict or direct edit of a dict

Reply:
Fair enough, let's keep as is and see what happens.